app/services/bulk_update/recommendations_uploads/create_csv_with_errors.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module BulkUpdate
+  module RecommendationsUploads
+    class CreateCsvWithErrors
+      include ServicePattern
+
+      def initialize(recommendations_upload:)
+        @recommendations_upload = recommendations_upload
+      end
+
+      def call
+        table = download_csv
+
+        # The first row of `table` is the 'do not edit' row. The second row of
+        # table is an actual trainee row, and corresponds to csv row number 3.
+        table.each.with_index(2) do |row, row_number|
+          row["Errors"] = errors_for_row(row_number)
+        end
```

Comment:
either I'm misunderstanding what this is doing, or you are misunderstanding `with_index` - when you pass it an int it uses that to offset the index count (not the index it self) e.g.:

<img width="508" alt="Screenshot 2023-03-16 at 08 53 36" src="https://user-images.githubusercontent.com/6339025/225565006-574047cd-90a7-4c8a-bf8c-68257a1b1976.png">

so this would still loop over the "do not edit" row as far as I can tell